### DISCUSSION TITLE
Contextual menu does not appear in thumbnails.

### DISCUSSION THREAD

**Discussion by kazdonkai** [RyotaUshio/obsidian-pdf-plus#55] [Q&A] [answered]
現在、PDFのサムネールを右クリックすると、コンテクストメニューがでる仕様になっていると思うのですが、画像のようになってしまい、表示できません。
何かのプラグインのせいかも知れませんが、何かヒントはありますか？
![IMG_3528](https://github.com/RyotaUshio/obsidian-pdf-plus/assets/90127278/abb67f5b-1cd5-4192-934a-ae970427996e)

**Comment by RyotaUshio** [comment]
ご質問ありがとうございます。

前提として、サムネイルを右クリックでコンテクストメニューが出るのは Obsidian そのものの仕様です。PDF++は、もともと存在するそのコンテクストメニューをより強力なものに置き換えるという立ち位置です(ただし、PDFファイルの変更を許可していない場合は特に変化はありません)。

それで本題ですが、PDF++によってその画像の状態になるというのは少し考えにくいかなと思います。
カーソルが虫眼鏡の形になるのはなんらかのCSS（テーマ・プラグイン・スニペット）のせいの可能性もあります。

- PDF++をオフにしてからObsidianを再起動すると、その症状は治りますか？
- もし治らない場合は、現在通常の vault で有効にしているプラグインやテーマ、または CSS snippets が悪さをしている可能性が高いです。この場合は、プラグイン・テーマを最新版にアップデートし、いったんすべて無効化してから**Obsidianを再起動**して、ひとつひとつ有効化することによって原因を突き止めるのが地道ですがいちばんの近道になります。

以下の情報を提供していただけると、もしかしたらお手伝いできるかもしれません。

- コマンドパレットから "Show debug info" コマンドを実行し、結果をコピー&ペースト
- (PDF++がオンのときだけ問題が起こる場合) 同じくコマンドパレットから "PDF++: Copy debug info" を実行し、クリップボードにコピーされた文字列をペースト（引用符などの文字が変化しないようにコードブロックで囲ってください）

**Comment by kazdonkai** [comment, marked as answer]
解決しました。
Obsidian CEOの@kepano によるMinimal テーマのせいでした。
他のテーマでは、コンテクストメニューが表示されました。

このテーマは、一番使われている準標準テーマなので、どこかに告知しておくなり、@kepanoに連絡するなり何か対応をとられた方が良いと思います。

**Comment by RyotaUshio** [comment]
Minimal にバグレポートを提出しました。→ Minimal 7.5.2 で修正済み

https://github.com/kepano/obsidian-minimal/issues/702

（Minimalのレポジトリにバグレポートを出すだけにして、他の場所ではあまり直接メンションしないほうがいいかもしれないです）

また、READMEにも注意書きを加えました。

**Comment by kazdonkai** [comment]
他の事ですが、PDFのサムネールに関する質問があるので、ここでします。
モバイル版(iOS)で長押しをするとコンテクストメニューが表示されますが、画像のようになり、PDF++で表示されるべきメニューと違うと思うのですが・・
![IMG_3529](https://github.com/RyotaUshio/obsidian-pdf-plus/assets/90127278/edb80fc8-8498-412d-a0b9-1c72cf538d52)

**Reply by RyotaUshio** [reply]
https://github.com/RyotaUshio/obsidian-pdf-plus/discussions/55#discussioncomment-8577486 で

> ただし、PDFファイルの変更を許可していない場合は特に変化はありません

といいましたが、以下の設定がオンになっていない場合（デフォルトはオフ）、PDF++は既存のPDFファイルの変更を伴う処理を行いません。

<img width="794" alt="image" src="https://github.com/RyotaUshio/obsidian-pdf-plus/assets/72342591/033a64e4-2e5c-4090-bc7b-6b43a3cd8c04">

結果として、サムネイルを右クリックしたときにも、Obsidianデフォルトと見た目は同じコンテキストメニューが表示されます（ただし、Obsidianデフォルトのものとは異なり、コピーされるリンクのdisplay textの形式はツールバーのドロップダウンメニューから指定したり、プラグインの設定画面からカスタマイズしたりできます）。

この設定をオンにして、annotation authorの欄を埋めるとPDFファイルの変更が可能になり、コンテキストメニューにもファイルの変更を伴う、よりadvancedな項目が複数追加されるようになります。

<img width="402" alt="image" src="https://github.com/RyotaUshio/obsidian-pdf-plus/assets/72342591/86b56938-ddb2-42c7-8874-9172969ab044">

（次回からは、類似の話題でも新しくスレッドを作っていただけるとありがたいです。そのほうが他のユーザが後から探しやすいと思うので）

**Comment by kazdonkai** [comment]
説明して頂いた通りに設定した所うまく表示されました。
ありがとうございます。
これでこのスレッドは終了します。
